README.md

```diff
@@ -1,15 +1,44 @@
+[![Review Assignment Due Date](https://classroom.github.com/assets/deadline-readme-button-24ddc0f5d75046c5622901739e7c5dd533143b0c8e959d652212380cedb1ea36.svg)](https://classroom.github.com/a/FgMJElkj)
 # Theory vs. Practice
 
 - List 3 reasons why asymptotic analysis may be misleading with respect to
   actual performance in practice.
+  - constant factors and hidden constants typically in asymtotic analysis constant factors are ignored.
+    In real life situation these factors can have a decent impact on the actual time comlexity.
+  - The type of analysis you are doing. If it is big O notation then this would give you loose bounds which mean that the runtime my be 
+    actually faster than the analysis is for the worst case scenario.
+  - Hardware may have different instruction sets algorithms that perform well on one machine may not perform the same on another type 
+    of machine. Asymtoic analysis does not take this into consideration and can lead to misleading expectations of performance.
 
 - Suppose finding a particular element in a binary search tree with 1,000
   elements takes 5 seconds. Given what you know about the asymptotic complexity
   of search in a binary search tree, how long would you guess finding the same
   element in a search tree with 10,000 elements takes? Explain your reasoning.
+  - Given that the BST is balanced we can assume that the average case time complexity is $\ O(\log_2 n) $
+  - $\(\log_2 n)$ is how we can determine the hieght of the tree with n being the amount of elements in the tree.
+  - we also know that with the hieght of the tree we can determine the maximum amount of comparisons that would be made to find a
+    particular element in the tree.
+  - $\\log_2 1000 = 9.965$ which we can just round to 10 given the tree is balanced. So know we know the height of the tree is 10
+    and the max number of comparisons made is 11 or $\\log_2 (n) + 1$
+  - So now we know a tree that has 1000 elements has a height of ten and a max number of comparisons of 11. We also know this takes 5 
```

Comment:
Each node in the tree represents a comparison, so how do you get to 11 comparisons?

Reply:
For some reason in my head I was adding an extra comparison on because I was thinking of it like if we get to the bottom of the tree we still need to check if the value is equal to our target. Which would be extra but thinking about it now, it does make sense that there would be and extra comparison.

Reply:
Doesn't make sense